File: playwright-project/src/config/userContext.ts
import { Role } from '@constants/role';

export class UserContext {
    token?: string;

    constructor(
        public username: string,
        public password: string,
        public role: Role = Role.ADMIN,
        public email: string = username,
    ) {
        this.username = username;
        this.password = password;
        this.role = role;
        this.email = email;
    }
}

File: selenium-project/src/main/java/com/master/seleniumproject/models/UserContext.java
package com.master.seleniumproject.models;

public class UserContext {
     private String username;
     private String password;
     private String token;

     public UserContext(String username, String password) {
          this.username = username;
          this.password = password;
     }

     public String getUsername() {
          return username;
     }

     public String getPassword() {
          return password;
     }

     public String getToken() {
          return token;
     }

     public void setToken(String token) {
          this.token = token;
     }
}
